src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -231,7 +266,139 @@ export const handleEncryptedSubmission: RequestHandler = async (
       myinfoResult.value
   }
 
-  return next()
+  // Encrypt Verified SPCP Fields
+  if (form.authType === AuthType.SP || form.authType === AuthType.CP) {
+    const encryptVerifiedContentResult = VerifiedContentFactory.getVerifiedContent(
+      { type: form.authType, data: res.locals },
+    ).andThen((verifiedContent) =>
+      VerifiedContentFactory.encryptVerifiedContent({
+        verifiedContent,
+        formPublicKey: form.publicKey,
+      }),
+    )
+
+    if (encryptVerifiedContentResult.isErr()) {
+      const { error } = encryptVerifiedContentResult
+      logger.error({
+        message: 'Unable to encrypt verified content',
+        meta: logMeta,
+        error,
+      })
+
+      // Passthrough if feature is not activated.
+      if (error instanceof MissingFeatureError) {
+        return next()
+      }
+
+      return res
+        .status(StatusCodes.BAD_REQUEST)
+        .json({ message: 'Invalid data was found. Please submit again.' })
+    }
+  }
+
+  // Verify Structure of Encrypted Response
+  // Step 1: Add req.body.encryptedContent to req.formData
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  ;(req as WithFormData<typeof req>).formData = req.body.encryptedContent
+  // Step 2: Add req.body.attachments to req.attachmentData
+  ;(req as WithAttachmentsData<typeof req>).attachmentData =
+    req.body.attachments || {}
+
+  // Save Responses to Database
+  const { formData, attachmentData } = req
+  const { verified } = res.locals
+  const attachmentMetadata = new Map()
+  const attachmentUploadPromises = []
+
+  // Object.keys(attachmentData[fieldId].encryptedFile) [ 'submissionPublicKey', 'nonce', 'binary' ]
+  for (const fieldId in attachmentData) {
+    const individualAttachment = JSON.stringify(attachmentData[fieldId])
+
+    const hashStr = crypto
+      .createHash('sha256')
+      .update(individualAttachment)
+      .digest('hex')
+
+    const uploadKey =
+      form._id + '/' + crypto.randomBytes(20).toString('hex') + '/' + hashStr
+
+    attachmentMetadata.set(fieldId, uploadKey)
+    attachmentUploadPromises.push(
+      AwsConfig.s3
+        .upload({
+          Bucket: AwsConfig.attachmentS3Bucket,
+          Key: uploadKey,
+          Body: Buffer.from(individualAttachment),
+        })
+        .promise(),
+    )
+  }
```

Comment:
can help to see if it's possible to extract some of this detail into a service? we're trying to keep things simple in the controller so that no real work is performed here; it should only be determining HTTP response statuses

Reply:
Maybe on the next PR? I don't want to do a bunch of things at the same time :p Adding a TODO